### PR TITLE
Inner block slider component

### PR DESCRIPTION
**Pull request by mattheu**
The inner block slider component is a utility component for managing sliders or carousels within the block editor. It uses inner blocks for each slide. It provides no front end JS or even any markup at all, so you can integrate it with whatever  front end implementation you choose.  

This is adapted from a 10up utility component. I couldn't actually get that working, so I fixed it as well as refectoring and modernizing it a bit. 

This is currently in use on another project.

### CODE REVIEW

File: js/inner-block-slider/index.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+import { useSelect, useDispatch } from '@wordpress/data';
+
+import Navigation from './navigation';
+
+/**
+ * InnerBlockSlider component.
+ *
+ * @param {object} root0 - Component props.
+ * @param {string} root0.parentBlockId - Parent block clientId.
+ * @param {number} root0.slidesPerPage - Number of visible slides.
```
Comment (gonzomir) [inline js/inner-block-slider/index.js:15]: This is not used any more, only one slide is visible at a time. Should this param be removed?

File: js/inner-block-slider/index.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	);
+};
+
+InnerBlockSlider.defaultProps = {
+	template: null,
+	slideLimit: 5,
+};
+
+InnerBlockSlider.propTypes = {
+	slidesPerPage: PropTypes.number,
```
Comment (gonzomir) [inline js/inner-block-slider/index.js:107]: Maybe remove if here too.

**Review by gonzomir — CHANGES_REQUESTED** [review]
Great go see this component introduced here! Only one small thing - the `slidesPerPage` prop doesn't do anything and can be removed.

**Comment by mattheu** [comment]
Thanks @gonzomir did a bit of cleanup.

File: js/inner-block-slider/index.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+InnerBlockSlider.defaultProps = {
+	template: null,
+	slideLimit: 5,
+};
+
+InnerBlockSlider.propTypes = {
+	parentBlockId: PropTypes.string.isRequired,
+	allowedBlock: PropTypes.string.isRequired,
+	template: PropTypes.array,
+};
```
Comment (gonzomir) [inline js/inner-block-slider/index.js:108]: Oh, one more thing, should `slideLimit` appear also here?

**Review by gonzomir — APPROVED** [review]

**Comment by mattheu** [comment]
This ended up being included in the [HM Block Editor Components library](https://github.com/humanmade/block-editor-components/tree/main/src/components/InnerBlockSlider) instead.

**Pull request closed without merging**